### ISSUE TITLE
[Bug] Namco Museum Battle Collection: Major audio issues with cracking and skipping

### ISSUE THREAD

**Issue by Goldenfreddy0703** [hrydgard/ppsspp#15485]
### Game or games this happens in

Namco Museum Battle Collection

### What area of the game / PPSSPP

When I play the Original Games, the audio starts to crackle which causes issues.

- I have tried multiple devices to see if its the PPSSPP itself, Its an issue for every device i tried.
- I have tried every version of Namco Museum Battle Collection including all languages and still same audio issue. 
- From all the other original games, Mappy is the only game that has slow audio for some odd reason
- All the Arrangement Games seem to be fine, its only Original Games. 

Here is a screenshot of the Original Games. 
 ![Original](https://user-images.githubusercontent.com/62456796/163726742-883e339e-46f2-4758-a7b9-443a94c83615.png)

I do not have a savedata for this because this is effected to multiple devices.  

### What should happen

The Original Games should not have Major audio issues with cracking and skipping. It Should have Normal audio like the Arrangement Games. This could be due to Simulating Block Transfer Effects but i doubt it. It is an important feature that is needed to run the Original Games. 
![Simulating Block Transfer Effects](https://user-images.githubusercontent.com/62456796/163727049-0c05cde5-21f4-4c44-8815-ceb8832f3f1e.png)

### Logs

I had trouble uploading a debug log but i hope this will help. 
[PPSSPP Log.txt](https://github.com/hrydgard/ppsspp/files/8502229/PPSSPP.Log.txt)

### Platform

Windows

### Mobile phone model or graphics card

Nvidia GeForce GTX 1660 Ti with Max-Q Design

### PPSSPP version affected

v1.12.3

### Last working version

None

### Graphics backend (3D API)

Direct3D 11

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by ANR2ME** [comment]
Your logs file didn't show much info as it only logs Error message, you should run the `Debug Log.bat` file to get more detailed logs.

**Comment by Goldenfreddy0703** [comment]
Should i use the Lastest Git Build and run the Debug Log.bat?

**Comment by ANR2ME** [comment]
> Should i use the Lastest Git Build and run the Debug Log.bat?

Yes, preferably getting the logs from the latest dev build.

**Comment by Goldenfreddy0703** [comment]
Ok i will do that, i am testing it now.

**Comment by Goldenfreddy0703** [comment]
Hey so apparently,  i have a log but i can't upload it, github is say not included in list

**Comment by ANR2ME** [comment]
> Hey so apparently, i have a log but i can't upload it, github is say not included in list

zip the log file first

**Comment by Goldenfreddy0703** [comment]
Thank you, sorry that took forever
[ppsspplog.zip](https://github.com/hrydgard/ppsspp/files/8502537/ppsspplog.zip)

**Comment by hrydgard** [comment]
Thanks for the log, not sure it's gonna help too much in this case though :/  Needs investigation for sure, I'm guessing these games do their own mixing and there's probably some obscure audio buffering issue that they run into, that normal games don't because they use the PSP's native mixer API.

**Comment by Goldenfreddy0703** [comment]
True, i always thought it could be the Simulating Block Transfer Effects cause the original games need that but im not too sure, it could be due to audio with that function but im not fully sure. One thing i herd from a thread is maybe changed the audio format which could possibly help but im not sure if that would work

**Comment by Goldenfreddy0703** [comment]
Hey, I was looking at other issues and pr, would this issue relate to this one by anychance?
https://github.com/hrydgard/ppsspp/issues/5213

**Comment by unknownbrackets** [comment]
You mentioned trying other devices, more detail on that might help.

Some specific things to try:
 * Try switching the audio backend from WASAPI to DirectSound.
 * Enable the FPS display (set to Both), and check if it's showing < 100% at any time during gameplay.
 * Enable "Show developer menu" under Settings -> Tools -> Developer tools, then during the game click "DevMenu" in the top corner.  In the menu that appears, scroll down if needed and check "Show frametimes graph".  Then reproduce the issue and post a screenshot with the frametimes displayed (we're looking for spikes.)
 * Enable the setting "Log dropped frame statistics" under Settings -> Tools -> Developer tools, then use the info log method again to get frame drop statistics.
 * Try other graphics backends than Direct3D 11, specifically OpenGL or Vulkan.
 * Try enabling the software rendering setting, but only using a very recent git build not v1.12.3.

If there's no spikes, no logged frame drops, it's always 100%, and changing the backends does nothing - then this must be an emulation issue in audio mixing.  But if any of these help, it may be some sort of performance issue.

-[Unknown]

**Comment by Goldenfreddy0703** [comment]
It's got to be an emulation issue at this point because I've tried it on multiple devices such as android tv, windows, Linux, my android phone, through emulated devices, ect.

**Comment by unknownbrackets** [comment]
Well, it could be a performance problem (I mean, still a bug in PPSSPP) that's happening on all these devices.  My questions are designed to ferret out a performance problem.

If it's happening on multiple types of devices like that, I wouldn't bother with the graphics backends or audio backends, but the other things are still useful.

-[Unknown]

**Comment by Goldenfreddy0703** [comment]
Your right, I will try some of the specifics and get you a log for it so it could help more.

**Comment by Goldenfreddy0703** [comment]
Hey so i tried all the specific things and i think i got you an infolog to help.
Im not too sure if this is an info log but here is a log from the latest branch. Here is also another debug log to help too. 
InfoLog
[InfoLog.zip](https://github.com/hrydgard/ppsspp/files/8555011/InfoLog.zip)
DebugLog
[DebugLog.zip](https://github.com/hrydgard/ppsspp/files/8555099/DebugLog.zip)

**Comment by Goldenfreddy0703** [comment]
Hey so i just wanted to come back and check on this, did this information help you guys out on this bug?

**Comment by Goldenfreddy0703** [comment]
Hey so its been a few days and i wanted to see how this is going, any new information on this by anychance?

**Comment by Goldenfreddy0703** [comment]
Have you tried Paxman or Galaga original?

**Comment by Goldenfreddy0703** [comment]
Hey guys, i know its been almost a month and i understand if you guys are busy with other issues but any information on this one by anychance?

**Comment by unknownbrackets** [comment]
Well, unfortunately the log didn't help outside iirc showing a couple drops from using software rendering.  You didn't mention any of those settings having an impact, so it probably is some uncommon api usage that'll need to be investigated.  I just double checked though and it is using sceSas (although mono atrac3+), so doesn't look too crazy.

Maybe it's doing something evil mucking with the actual Sas structs and double buffering.  Hopefully not.

-[Unknown]

**Comment by Goldenfreddy0703** [comment]
Awww thanks for telling me, I wish there was a way to fix the audio for the old games. Idk why audio is messed up, slow, fast, it could be just rendering but again, not too sure

Thank you tho.

**Comment by UnveilingCoyote** [comment]
When will this be fixed?

**Comment by Goldenfreddy0703** [comment]
I was wondering about this as well, glad this is on the v1.17.0 milestone tho. 
Thank you

**Comment by UnveilingCoyote** [comment]
Update on this?

**Comment by UnveilingCoyote** [comment]
Will this be fixed anytime soon?

**Comment by Goldenfreddy0703** [comment]
Hey there, been awhile sense I responded to this issue but is ppsspp already past the 1.17.0 milestone? Its probably been about 2 years and im surprised this is still an ongoing issue.

**Comment by hrydgard** [comment]
1.17 is the upcoming milestone, we're not past it.

And please don't post just for updates. I have looked at it a couple of times and haven't been able to figure it out. I will look again but no guarantees I'll solve it before the 1.17 release.

Feel free to help debug it.

**Comment by Goldenfreddy0703** [comment]
Hey sure and it's totally fine. I think awhile back, it had something to do with the audio format but I'm not too sure. Maybe I will look into this more and help with debugging soon.

Thank you

**Comment by sum2012** [comment]
@Goldenfreddy0703 Can you record the cracking and skipping (include graphic and sound) into https://www.youtube.com (and share the link with us) ?

**Comment by UnveilingCoyote** [comment]
Here is the slowdown

Uploading Screen_Recording_20240117_173022_PPSSPP.mp4…

**Comment by UnveilingCoyote** [comment]
https://github.com/hrydgard/ppsspp/assets/115852613/f3d1ce22-db85-4510-83ff-3ba0b8f78065

**Comment by Goldenfreddy0703** [comment]
Hey @sum2012 it's been awhile sense I used ppsspp but I can try to get on my computer later and see if I can record it for you. Thank you for pinging me.

**Comment by sum2012** [comment]
@UnveilingCoyote I cannot see your video

**Comment by UnveilingCoyote** [comment]
Update on this?
